### DISCUSSION TITLE
[CLOSED] Suggestion about a built-in PDF viewer for Bromite.

### DISCUSSION THREAD

**Discussion by bat999** [bromite/bromite#1864] [Ideas]
Hi
There is a Firefox addon "**Android PDF.js**".
https://addons.mozilla.org/en-US/android/addon/android-pdf-js/

It works OK with ANDROID Firefox Nightly, Fennec F-Droid, Iceraven etc.
PDF files are viewed in the browser without downloading first.

Apparently this addon is built around "**PDF.js**".
https://github.com/mozilla/pdf.js

Could **Bromite** be built with this included?

I'm putting this in the "Discussions" section.
I realize that it won't improve privacy or security.
Maybe it would improve productivity or convenience.
If the dev/collaborators think it's no good then please shut this thread down.
If the dev/collaborators think it's OK I will open a Feature Request.

**Comment by ghost** [comment]
you can use GrapheneOS PDF viewer which Uses the Mozilla's pdf.js

**Comment by bat999** [comment]
Hi

**JustAnUserLol
"you can use GrapheneOS PDF viewer which Uses the Mozilla's pdf.js"**

That guy didn't read my question!

Anyway, there's another project called "pdfium" by Google.
https://github.com/chromium/pdfium
Maybe this would build easier in a chromium-based browser such as Bromite.

**Comment by bat999** [comment]
Hi
I looked into this again.

Firefox browser for Linux Desktop.
PDF files open in the built-in viewer.
It uses pdfjs.

Chromium browser for Linux Desktop.
PDF files open in the built-in viewer.
It uses PDFium.

Mozilla-based browsers for ANDROID phone.
PDF files can be opened directly with a third-party addon such as **Android PDF.js**.

Chromium-based browsers for ANDROID phone.
PDF files are downloaded then they can be viewed (automatically) with a third-party app such as **MuPDF.**
This method works OK but the downloaded PDF files then have to be manually deleted.

I think it would be great to have a built-in PDF viewer for Bromite.
But it will probably entail a lot of work.

Maybe someday they will bundle PDFium in with the Chromium code for ANDROID.

Please consider this thread/discussion CLOSED.
